### DISCUSSION TITLE
Show LM on the current space (virtual MacOS desktops)

### DISCUSSION THREAD

**Discussion by faressoft** [LaunchMenu/LaunchMenu#190] [Q&A] [answered]
Is there an option to enable LM to be shown on the current active space on MacOS spaces instead of switching to the space where LM is launched from the beginning?

If such a feature is not implemented, could you please inform me where I can find the handler (in the code of LM) that displays the window when I press the following keys:

<img width="668" alt="Screenshot 2023-09-11 at 23 11 09" src="https://github.com/LaunchMenu/LaunchMenu/assets/1908658/e020542f-116e-47f2-9e2c-ff67fba4fb42">

**Comment by sancarn** [comment]
I've been informed that this is namely controlled by [`macVisibilityController`](https://github.com/LaunchMenu/LaunchMenu/blob/development/packages/core-applets/window-manager/src/visibility/OScontrollers/macVisibility.ts). Whether Electron provides APIs to bring the app to the active space however is outside of our knowledge.

I would generally be in favor of this though, as I think it makes sense that LM is sytem wide, not specific to the space. I think it's also worthwhile reflecting this feature on Windows OS's desktops feature.

**Reply by sancarn** [reply, marked as answer]
With a bit of investigation this is accomplishable using `browserWin.setVisibleOnAllWorkspaces(true);` - I suggest this would be best added with a setting :)

**Reply by faressoft** [reply]
@sancarn Thanks a lot. It works.
